grammar/Model.g4

```diff
@@ -5,21 +5,21 @@ import Expr;
 model: ( subscriptRange | equation )+ ;
 
 // A subscript range definition names subscripts in a dimension.
-subscriptRange : ( Id ':' ( subscriptList | subscriptSequence | expr ) subscriptMappingList? ) | ( Id '<->' Id ) ;
+subscriptRange : ( ( Id ':' ( subscriptList | subscriptSequence | expr ) subscriptMappingList? ) | ( Id '<->' Id ) ) unitsDoc* ;
 subscriptSequence : '(' Id '-' Id ')' ;
 subscriptMappingList : '->' subscriptMapping ( ',' subscriptMapping )* ;
 subscriptMapping : Id | '(' Id ':' subscriptList ')' ;
 
 // An equation has a left-hand side and a right-hand side.
 // The RHS is a formula expression, a constant list, or a Vensim lookup.
 // The RHS is empty for data equations.
-equation : lhs ( ( ':=' | '==' | '=' ) ( expr | constList ) | lookup )? ;
+unitsDoc : '~' .*? '|' ;
+equation : lhs ( ( ':=' | '==' | '=' ) ( expr | constList ) | lookup )? unitsDoc* ;
 lhs : Id ( '[' subscriptList ']' )? ':INTERPOLATE:'? ( ':EXCEPT:' '[' subscriptList ']' ( ',' '[' subscriptList ']' )* )? ;
 
 // The lexer strips some tokens we are not interested in.
 // The character encoding is given at the start of a Vensim file.
 // The units and documentation sections and group markings are skipped for now.
```

Comment:
This comment ("The units and documentation sections and group markings are skipped for now.") needs to be updated now that we're no longer skipping units/docs.

Reply:
The `unitsDoc` expression may match a large amount of text, as some of the comments are lengthy. Now that they're not simply skipped, the parser may be taking a lot of time and memory handling them.

I don't see how to resolve the bug in this issue without splitting equations on this expression. Maybe there's a way to include it in the equation and subscriptRange expressions and still discard the contents. I will find out.

Reply:
Now that we're back to skipping `unitsDoc`, the code comment is once again accurate, so my review comment above is now obsolete.  Marking this conversation as resolved.